packages/ui/src/components/chat.tsx

```diff
@@ -0,0 +1,149 @@
+import { ChangeEvent, KeyboardEvent, ReactNode, RefObject } from 'react'
+
+import { Button, Icon, Input, Text } from '@/components'
+import { cn } from '@/utils/cn'
+
+// Root Container
+const Root: React.FC<{ children: ReactNode }> = ({ children }) => {
+  return <div className="w-full h-full overflow-y-auto flex flex-col">{children}</div>
+}
+
+// Body
+const Body: React.FC<{ children: ReactNode }> = ({ children }) => {
+  return <div className="flex-1 flex flex-col gap-6 mb-3 p-6">{children}</div>
+}
+
+// Footer
+const Footer: React.FC<{ children: ReactNode }> = ({ children }) => {
+  return <div className="sticky bottom-0 p-6 pt-0 bg-background">{children}</div>
+}
+
+// Message Component
+interface MessageProps {
+  self?: boolean
+  time?: string
+  avatar?: ReactNode
+  actions?: ReactNode
+  children: ReactNode
+}
```

Comment:
Instead of manually typing `children`, you can use `PropsWithChildren`.
```suggestion
interface MessageProps extends PropsWithChildren {
  self?: boolean
  time?: string
  avatar?: ReactNode
  actions?: ReactNode
}
```